torch/csrc/CudaIPCTypes.cpp

```diff
@@ -0,0 +1,112 @@
+#include <torch/csrc/CudaIPCTypes.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <iostream>
+#include <map>
+
+CudaIPCReceivedData::CudaIPCReceivedData(std::shared_ptr<void> shared_ptr)
+    : shared_ptr_(std::move(shared_ptr)) {}
+
+int64_t CudaIPCSentData::get() {
+  return *counter_ptr;
+}
+
+CudaIPCSentDataLimbo::~CudaIPCSentDataLimbo() {
+  collect();
+  if (shared_blocks_.size() > 0) {
+    AT_ERROR(
+        "Producer process has been terminated before all shared CUDA tensors released.");
+  }
+}
+
+void CudaIPCSentDataLimbo::collect() {
+  std::vector<CudaIPCSentData*> kept_blocks;
+  for (auto const& sd : shared_blocks_) {
+    if (sd->get() > 0) {
+      // std::cout << "Collect keeping in limbo " << sd->get() << " " <<
```

Comment:
There are many commented pieces of code that looks like debug logs. Clean up? 
same for lines 39-40, 48-49, 54-55, 58-59, 73, 83 in this file and others.